src/libostree/ostree-repo-pull.c

```diff
@@ -1534,11 +1498,13 @@ process_one_static_delta_fallback (OtPullData   *pull_data,
         {
           if (!g_hash_table_lookup (pull_data->requested_metadata, checksum))
             {
-              gboolean do_fetch_detached;
               g_hash_table_insert (pull_data->requested_metadata, checksum, checksum);
               
-              do_fetch_detached = (objtype == OSTREE_OBJECT_TYPE_COMMIT);
-              enqueue_one_object_request (pull_data, checksum, objtype, do_fetch_detached, FALSE);
+              /* Fetch metadata */
+              if (objtype == OSTREE_OBJECT_TYPE_COMMIT)
+                fetch_object (pull_data, g_strdup (checksum), OSTREE_OBJECT_TYPE_COMMIT_META);
```

Comment:
`(transfer full)` input functions are pretty unusual; I can't think of any other ones in the ostree source, i.e. `git grep '(transfer full)' |grep -v Returns |grep -v '(out)'` is empty.  There are some in GLib but still quite rare.

Why not just keep the memory owned by the hash table like it was originally and drop the strdup?